pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -120,13 +130,36 @@ func (t *Terminator) Evict(pods []*v1.Pod) {
 	// b. non-critical daemonsets
 	// c. critical non-daemonsets
 	// d. critical daemonsets
-	if len(nonCriticalNonDaemon) != 0 {
-		t.evictionQueue.Add(nonCriticalNonDaemon...)
-	} else if len(nonCriticalDaemon) != 0 {
-		t.evictionQueue.Add(nonCriticalDaemon...)
-	} else if len(criticalNonDaemon) != 0 {
-		t.evictionQueue.Add(criticalNonDaemon...)
-	} else if len(criticalDaemon) != 0 {
-		t.evictionQueue.Add(criticalDaemon...)
+	t.EvictInOrder([][]*v1.Pod{
+		nonCriticalNonDaemon,
+		nonCriticalDaemon,
+		criticalNonDaemon,
+		criticalDaemon,
+	})
+}
+
+func (t *Terminator) EvictInOrder(pods [][]*v1.Pod) {
+	for _, podList := range pods {
+		if len(podList) > 0 {
+			// evict the first list of pods that is not empty, ignore the rest
+			t.evictionQueue.Add(nil, podList...)
+			return
+		}
+	}
+}
+
+// if a pod should be deleted to give it the full terminationGracePeriodSeconds of time before the node will shut down, return the time the pod should be deleted
+func (t *Terminator) shouldDeletePodToEnsureGracePeriod(nodeGracePeriodExpirationTime *time.Time, pod *v1.Pod) *time.Time {
+	// check if pod is already terminating
+	if !pod.DeletionTimestamp.IsZero() {
```

Comment:
If pod is already terminating, I don't think that we even throw it into the queue since we filter out terminating pods as part of our eviction call on L88 in the old version of the code

Reply:
Defensive coding habits.